app/controllers/reportings_controller.rb

```diff
@@ -197,18 +199,7 @@ def new
   end
 
   def create
-    @reporting = @project.reportings_via_source.build
-    @reporting.reporting_to_project_id = params['reporting']['reporting_to_project_id']
-
-    if @reporting.reporting_to_project.nil?
-      flash.now[:error] = l('timelines.reporting_could_not_be_saved')
-      render action: :new, status: :unprocessable_entity
-      return
-    end
-
-    check_visibility
-
-    if @reporting.save
+    if @reporting.reporting_to_project.present? && @reporting.project.visible? && @reporting.save
```

Comment:
The new, create and edit actions of reporting are limited to users having the edit_reportings permision:

```
    timelines.permission :edit_reportings,
                         { reportings: [:new, :create, :edit,
                                        :update, :available_projects] },
```

This is enforced by the before filter set for the controller
```
before_filter:  authorize
```

So without having tested it, I think the permissions are checked ok. At least for as long, as we do permission checking on the controller layer.